.github/scripts/test_and_publish.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python3
+
+import subprocess, sys, os, tempfile
+from distantrs import Invocation
+from glob import glob
+
+INVOCATION_DATA_PATH = '/tmp/distant-rs-invocation.txt'
+
+# required to find listener.py
+this_path = os.path.abspath(os.path.dirname(__file__))
+test_script_path = f"{os.environ.get('GITHUB_WORKSPACE')}/renode/test.sh"
+inv_data_str = ""
+robot_name = glob("*.robot")[0].split(".")[0]
+
+with open(INVOCATION_DATA_PATH, 'r') as f:
+    inv_data_str = f.read()
+
+inv_data_lst = inv_data_str.split("--")
+
+i = Invocation(
+        invocation_id=inv_data_lst[0],
+        auth_token=inv_data_lst[1]
+        )
+
+i.announce_target(robot_name)
+
+tmp = tempfile.NamedTemporaryFile()
+with open(tmp.name, 'w') as log:
+    # run the actual testing procedure
+    subprocess_args = [
+            test_script_path, 
+            "--listener",
+            os.path.join(this_path, f"results_listener.py:{inv_data_str}")
+                ]
+    subprocess_args.extend(sys.argv[1:])
+
+    process = subprocess.Popen(
+        subprocess_args,
+        stdout=subprocess.PIPE, 
+        stderr=subprocess.STDOUT, 
+        env=os.environ.copy()
+        )
+
+    ret = 0
+
+    while True:
+        line = process.stdout.readline().decode('utf-8').rstrip()
+        ret = process.poll()
+        if ret is not None:
+            break
+        if line:
+            log.write(f'{line}\n')
+            print(line)
+
+
+expected_summary_paths = [
+        'nunit_output.xml',
+        'robot_output.xml',
+        'log.html',
+        'report.html',
```

Comment:
please verify with the current list of artifacts. No nunit, but some images and a data dump file (this one can be big, up to 50megs. Is this a problem?)

Reply:
and these will be not "summary" but "artifacts" paths

Reply:
Done. I also reworked the `test_and_publish.py` script a bit so that it handles the execution of metrics analyzer and uploads its artifacts to the "fake" target. This way, all artifacts do end up in ResultStore.

Regarding bigger files -- this is not a problem, because the files themselves are uploaded to a Google Cloud Storage Bucket (so only the link to the file is stored in the invocation).